### PR TITLE
Feat/158/admin art create art for streetcode

### PR DESCRIPTION
**Pull request by VitaKotenko**
dev
## Summary of issue

Implemented ability create new art with adding streetcodes

## Summary of change

To add this functionality next items were implemented:

- created ArtCreateDto;
- added mapper for mapping ArtEntity -> ArtCreateDto and ArtCreateDto->ArtEntity;
- implemented command CreateArtCommand and handler CreateArtHandler;
- implemented validator CreateArtCommand that is used by ValidationPipelineBehavior to validate ArtCreateDto;
- added corresponding endpoint in ArtController;
- implemented CustomException;
- covered mediator with unit tests.

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=50%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  PR meets all conventions

### CODE REVIEW

File: Streetcode/Streetcode.BLL/MediatR/Media/Art/Create/CreateArtHandler.cs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+                .GetAllAsync(s => streetcodeIds.Contains(s.Id));
+
+            var streetcodeArts = streetcodes.
+                Select(sta => new StreetcodeArtEntity
+                {
+                    ArtId = newArt.Id,
+                    Art = newArt,
+                    StreetcodeId = sta.Id,
+                    Streetcode = sta
+                }).ToList();
```
Comment (jimmy2jk) [inline Streetcode/Streetcode.BLL/MediatR/Media/Art/Create/CreateArtHandler.cs:56]: To create `StreetcodeArt` it's enough to have StreetcodeId and ArtId. you don't need to retrieve `Streetcode` entities or provide `Art` entity to StreetcodeArt. Write to db operation will still work correctly but with less code and one db query less

File: Streetcode/Streetcode.BLL/MediatR/Media/Art/Create/CreateArtHandler.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+            var newArt = _mapper.Map<ArtEntity>(request.newArt);
+            if (newArt == null)
+            {
+                throw new CustomException("Cannot convert null to an art.", StatusCodes.Status204NoContent);
+            }
+
+            newArt.StreetcodeArts.Clear();
+            newArt = await _repositoryWrapper.ArtRepository.CreateAsync(newArt);
+            await _repositoryWrapper.SaveChangesAsync();
```
Comment (jimmy2jk) [inline Streetcode/Streetcode.BLL/MediatR/Media/Art/Create/CreateArtHandler.cs:42]: Save changes after all Create operation executed. You will still have ArtId for `StreetcodeArt`. But will not violate transactivity

**Review by jimmy2jk — CHANGES_REQUESTED** [review]
Fix issues provided in comments below. Overall, handler works correctly.

**Comment by VitaKotenko** [comment]
@jimmy2jk Everything is fixed. Thanks for your suggestions!

**Review by refokcer — APPROVED** [review]

**Pull request merged**